### PR TITLE
Feature/add config to main

### PR DESCRIPTION
**Pull request by MizukiSonoko**
## What is this pull request?
Add config loader to main
   
## Why do you implement it? Why do we need this pull request?
In initialize, we use config file
 
## Details/Features
List of features / major commits
- Add config loader to main
- Add test for config loader

### CODE REVIEW

File: irohad/main/iroha-main.cpp [hunk 1]
```diff
@@ -16,8 +16,19 @@ limitations under the License.
 
 #include <main/application.hpp>
 
+#include <common/config.hpp>
+#include <gflags/gflags.h>
+#include <cstring>
+#include "../../external/src/gflags_gflags-build/include/gflags/gflags.h"
```
Comment (lebdron) [inline irohad/main/iroha-main.cpp]: Not sure if this is required

File: test/module/irohad/common/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+
+add_executable(config_loader_test
```
Comment (lebdron) [inline test/module/irohad/common/CMakeLists.txt]: Please use `addtest` macro, defined in `cmake/functions.cmake`

File: irohad/main/iroha-main.cpp (hunk 1 above)
Comment (MizukiSonoko) [inline irohad/main/iroha-main.cpp]: ok

File: irohad/common/config.cpp [hunk 3]
```diff
@@ -24,44 +24,48 @@ limitations under the License.
...
+    }
 
-        std::string ConfigLoader::getStringOrElse(const std::string& key, std::string def){
-            if(doc[key.c_str()].IsString()){
-                return doc[key.c_str()].GetString();
-            }else{
-                return def;
-            };
-        }
+    std::string ConfigLoader::getStringOrElse(const std::string& key, std::string def) {
```
Comment (motxx) [inline irohad/common/config.cpp]: able to add const `, std::string def)` -> `, const std::string& def)`

File: irohad/common/config.cpp [hunk 4]
```diff
@@ -24,44 +24,48 @@ limitations under the License.
...
+    }
 
-        int ConfigLoader::getIntOrElse(const std::string& key, int def){
-            if(doc[key.c_str()].IsInt()){
-                return doc[key.c_str()].GetInt();
-            }else{
-                return def;
-            };
-        }
+    int ConfigLoader::getIntOrElse(const std::string& key, int def) {
```
Comment (muratovv) [inline irohad/common/config.cpp]: Naming is confusing. May be more suitable name like `getIntOrDefault`?

File: irohad/main/iroha-main.cpp (hunk 1 above)
Comment (muratovv) [inline irohad/main/iroha-main.cpp]: Could we fix this horrible include?

File: irohad/main/iroha-main.cpp [hunk 5]
```diff
@@ -16,8 +16,19 @@ limitations under the License.
...
+
+DEFINE_string(config, "iroha.conf", "Iroha's initialize config path");
```
Comment (muratovv) [inline irohad/main/iroha-main.cpp]: What is the purpose of `DEFINE_string`?

File: test/module/irohad/common/CMakeLists.txt (hunk 2 above)
Comment (muratovv) [inline test/module/irohad/common/CMakeLists.txt]: Andrew's  commentary is very important. Pls, rework cmake file with this feature. Example of usage see [here](https://github.com/hyperledger/iroha/blob/develop/test/module/irohad/torii/processor/CMakeLists.txt)

File: test/module/irohad/common/config_loader_test.cpp [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+class ConfigLoaderTest : public testing::Test {
+public:
+    virtual void SetUp() {
+        std::ofstream ofs("sample.conf");
+        ofs <<  "{\"name\":\"maruuchi\",\"age\":20, \"isRamen\": false}";
+        ofs.close();
+    }
+
+    virtual void TearDown() {
+      system("rm sample.conf");
```
Comment (muratovv) [inline test/module/irohad/common/config_loader_test.cpp]: `rm <file>` is very tricky. May be better solution to use `std::remove()`?

File: irohad/common/config.cpp (hunk 4 above)
Comment (takemiyamakoto) [inline irohad/common/config.cpp]: Yes, that's a better name

File: irohad/main/iroha-main.cpp (hunk 5 above)
Comment (l4l) [inline irohad/main/iroha-main.cpp]: I guess its purpose is in the help string. I guess that just didn't use yet

**Comment by l4l** [comment]
merged to #469

**Pull request closed without merging**